lib/pkg.ml

```diff
@@ -312,61 +285,33 @@ let distrib_file ~dry_run p =
       |> R.reword_error_msg (fun _ ->
              R.msgf "Did you forget to call 'dune-release distrib' ?")
 
-let distrib_user_and_repo p =
-  distrib_uri p >>= fun uri ->
-  let uri_error uri =
-    R.msgf
-      "Could not derive user and repo from opam dev-repo field value %a; \
-       expected the pattern $SCHEME://$HOST/$USER/$REPO[.$EXT][/$DIR]"
-      String.dump uri
-  in
-  match Text.split_uri ~rel:true uri with
-  | None -> Error (uri_error uri)
-  | Some (_, _, path) -> (
-      if path = "" then Error (uri_error uri)
-      else
-        match String.cut ~sep:"/" path with
-        | None -> Error (uri_error uri)
-        | Some (user, path) ->
-            let repo =
-              match String.cut ~sep:"/" path with
-              | None -> path
-              | Some (repo, _) -> repo
-            in
-            Fpath.of_string repo
-            >>= (fun repo -> Ok (user, Fpath.(to_string @@ rem_ext repo)))
-            |> R.reword_error_msg (fun _ -> uri_error uri) )
-
 let doc_uri p =
   opam_field_hd p "doc" >>| function None -> "" | Some uri -> uri
 
 let doc_dir = Fpath.(v "_build" / "default" / "_doc" / "_html")
 
-let doc_user_repo_and_path p =
-  doc_uri p >>= fun uri ->
-  (* Parses the $PATH of $SCHEME://$HOST/$REPO/$PATH *)
-  let uri_error uri =
-    R.msgf
-      "Could not derive publication directory $PATH from opam doc field value \
-       %a; expected the pattern $SCHEME://$USER.github.io/$REPO/$PATH"
-      String.dump uri
-  in
-  match Text.split_uri ~rel:true uri with
-  | None -> Error (uri_error uri)
-  | Some (_, host, path) -> (
-      if path = "" then Error (uri_error uri)
-      else
-        ( match String.cut ~sep:"." host with
-        | Some (user, g) when String.equal g "github.io" -> Ok user
-        | _ -> Error (uri_error uri) )
-        >>= fun user ->
-        match String.cut ~sep:"/" path with
-        | None -> Ok (user, path, Fpath.v ".")
-        | Some (repo, "") -> Ok (user, repo, Fpath.v ".")
-        | Some (repo, path) ->
-            Fpath.of_string path
-            >>| (fun p -> (user, repo, Fpath.rem_empty_seg p))
-            |> R.reword_error_msg (fun _ -> uri_error uri) )
+module Github = struct
+  let distrib_uri p =
+    distrib_uri p >>= fun uri ->
+    match Github_uri.Distrib.of_string uri with
+    | Ok uri -> Ok uri
+    | Error _ ->
+        R.error_msgf
+          "Could not derive user and repo from opam dev-repo field value %a; \
```

Comment:
The message is a bit misleading, how about something like:
```
One of homepage or dev-repo in xx.opam was expected to point to a github repository
```

Reply:
done